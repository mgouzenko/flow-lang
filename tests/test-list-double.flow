proc doubleGen(out double ochan){
  list <double> test = [2.2, 2.9, 80.0, .1];
  
  while(#test > 0) {
    @test -> ochan;
    test = ^test;
  }
  poison ochan;
}

proc printer(in double chan) {
  while(chan) {
    print_double(@chan);
    println();
  }
}

int main() {
  channel<double> chan;
  doubleGen(chan);
  printer(chan);
}