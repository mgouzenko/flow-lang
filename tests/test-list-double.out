2.2
2.9
80
0.1